src/libraries/System.Private.CoreLib/src/System/BitConverter.cs

```diff
@@ -608,6 +608,7 @@ public static float ToSingle(ReadOnlySpan<byte> value)
         /// </exception>
         /// <exception cref="ArgumentNullException"><paramref name="value"/> is null.</exception>
         /// <exception cref="ArgumentOutOfRangeException"><paramref name="startIndex"/> is less than zero or greater than the length of <paramref name="value"/> minus 1.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Does this method really need it? This method look small enough to get inlined naturally.

Reply:
I'm getting a JIT dump to check.

Reply:
Looks like this one is indeed small enough:
> INLINER: during 'fgInline' result 'success' reason 'below ALWAYS_INLINE size' for 'ConsoleApp50.Program:Main(System.String[])' calling 'System.BitConverter:ToDouble(System.Byte[],int):double'
> INLINER: during 'fgInline' result 'success' reason 'below ALWAYS_INLINE size'

The inner `ToInt64` fails to inline as it is not profitable.
```
Inline candidate looks like a wrapper method.  Multiplier increased to 1.
Inline candidate has an arg that feeds a constant test.  Multiplier increased to 2.
Inline candidate has arg that feeds range check.  Multiplier increased to 2.5.
Inline candidate callsite is boring.  Multiplier increased to 3.8.
calleeNativeSizeEstimate=801
callsiteNativeSizeEstimate=115
benefit multiplier=3.8
threshold=437
Native estimate for function size exceeds threshold for inlining 80.1 > 43.7 (multiplier = 3.8)
```

Which is a bit unfortunate as you might expect something like the following to be able to drop all the pre-checks:
```csharp
var bytes = new byte[1024];
new Random().NextBytes(bytes);
double d = BitConverter.Int64BitsToDouble(BitConverter.ToInt64(bytes, 0));
```